pydarn/io/dmap.py

```diff
@@ -1132,186 +915,26 @@ def __filename_check(self, filename: str = ""):
-                                                        rec_num,
-                                                        extra_fields)
-
-    # TODO: Do we want to check this? If not, then change
-    # SuperDARN_format_structure types to sets to get rid
-    # of dict_list2set method.
-    def incorrect_types_check(self, file_struct_list: List[dict],
-                              record: dict, rec_num: int):
-        """
-        Checks if the file structure fields data type formats are correct
-        in the record.
-
-        Parameters
-        ----------
-        file_struct_list : List[dict]
-            List of dictionaries representing the possible fields
-            in a file structure
-        record : dict
-            Dmap record
-        rec_num : int
-            Record number for a better error message information
-
-        Raises
-        ------
-        SuperDARNFileFormatError
-        """
-        complete_dict = {}
-        for file_struct in file_struct_list:
-            complete_dict.update(file_struct)
-        incorrect_types_check = {param: complete_dict[param]
-                                 for param in record.keys()
-                                 if record[param].data_type_fmt
-                                 != complete_dict[param]}
-        if len(incorrect_types_check) > 0:
-            raise pydmap_exceptions.SuperDARNDataFormatError(incorrect_types_check,
-                                                             rec_num)
-
-    def superDARN_file_structure_to_bytes(self, file_struct_list: List[dict]):
-        # TODO: might make rec_num a field in the class as it may be
-        # useful in error messages for both DmapRead and DmapWrite
-        for rec_num in range(len(self.dmap_records)):
-            record = self.dmap_records[rec_num]
-            # field checks
-            self.extra_field_check(file_struct_list, record, rec_num)
-            self.missing_field_check(file_struct_list, record, rec_num)
-            self.incorrect_types_check(file_struct_list, record, rec_num)
-            # start converting
-            self.__dmap_record_to_bytes(record)
-
     def dmap_records_to_bytes(self):
         """
         Loops through the dmap records and calls dmap_record_to_bytes
         to convert the dmap records to a byte array.
 
         Future use of this function is for parallelization.
         """
-        for record in self.dmap_records:
-            self.__dmap_record_to_bytes(record)
-
-    def __dmap_record_to_bytes(self, record: dict):
+        # For performance increase len of the records can be
+        # attribute value initialized in the class
+        pydarn_logger.debug("Converting DMAP records to bytes")
+        self.rec_num = 0
+        for self.rec_num in range(len(self.dmap_records)):
+            record = self.dmap_records[self.rec_num]
+            self._dmap_record_to_bytes(record)
+
+    def _dmap_record_to_bytes(self, record: dict):
```

Comment:
Are you referring to the encoding_identifier?

Reply:
Yeah. Also, I think that C dmap just referred to this variable as "code". did you find some information on this being an encoding?

Reply:
https://superdarn.github.io/rst/superdarn/src.doc/rfc/0006.html 
This is where it is noted so I took the terminology from here since "code" seems too vague to me.

Reply:
okay good to know.